esiclient/v1/node_network.py

```diff
@@ -264,21 +264,48 @@ def take_action(self, parsed_args):
         ironic_client = self.app.client_manager.baremetal
         neutron_client = self.app.client_manager.network
 
-        node = None
-        port = None
+        ports = list((port for port in ironic_client.port.list(detail=True)))
```

Comment:
If you want a list, just use a list comprehension instead of using generator expression and then transforming it into a list:

```
ports = [port for port in ironic_client.port.list(detail=True)]
```

Reply:
okay will do this.